---
title: "Examining allele effects for chromosome 19 traits"
author: "Frederick Boehm"
date: "10/30/2018"
output: github_document
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```

We first load the Rdata file from Data Dryad.

```{r}
load("../data/Attie_DO378_eQTL_viewer_v1.Rdata")
```

```{r}
foo <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 19, lod > 10)

(dat <- dataset.islet.rnaseq$annots %>% 
  filter(chr == 19) %>%
  right_join(foo, by = c("gene_id" = "annot.id")) %>%
  filter(chrom == 19, chr == 19) %>%
  arrange(desc(lod))
 )
```

We have `r nrow(dat)` local traits (with lods greater than 10) on chromosome 19.

Let's create a histogram of the QTL peak positions.

```{r}
hist(dat$pos)
```


*Asah2* has a LOD above 100 and is near the center of the chromosome. The four genes with greater LODs are not near the middle of the chromosome. It also looks like *Asah2* has some interesting biology, per its wikipedia page: https://en.wikipedia.org/wiki/ASAH2





```{r}
dat %>%
  filter(symbol == "Asah2")
asah2_id <- dat %>%
  filter(symbol == "Asah2") %>%
  select(gene_id) %>%
  unlist()
```

We can then isolate the *Asah2* expression values.

```{r}
asah2_expr <- dataset.islet.rnaseq$expr[ , colnames(dataset.islet.rnaseq$expr) == asah2_id, drop = FALSE]
```



Now, we save the 378 mice by 346 traits - including Asah2 - expression matrix for later use with Condor.

```{r}
trait_ids <- dat %>% 
  select(gene_id) %>%
  unlist()
```

```{r}
expr_chr19 <- dataset.islet.rnaseq$expr[ , 
                           colnames(dataset.islet.rnaseq$expr) %in% trait_ids, 
                           drop = FALSE]
```

Now, let's save the expression traits for only those traits with peak positions within 10 Mb of the peak position for *Asah2*.

```{r}
asah2_pos <- dat %>%
  filter(symbol == "Asah2") %>%
  select(pos) %>%
  unlist()
trait_ids_10mb <- dat %>% 
  filter(pos < asah2_pos + 10, pos > asah2_pos - 10) %>%
  select(gene_id) %>%
  unlist()

```

```{r}
expr_chr19_10mb <- dataset.islet.rnaseq$expr[ , 
                           colnames(dataset.islet.rnaseq$expr) %in% trait_ids_10mb, 
                           drop = FALSE]
```



## Examine the map object to see how many markers are spanned by the traits in chr19_expr_10mb.rds

```{r}
m19 <- map$`19`
m19_tib <- tibble(marker_id = names(m19), position = m19)
which(m19_tib$position > 22)[1]
which(m19_tib$position > 42)[1]
```

## Calculate allele effects for Asah2

```{r}
library(qtl2)
```

```{r}
s1c_asah2 <- scan1coef(genoprobs = genoprobs[, 19], pheno = asah2_expr, kinship = K$`19`, addcovar = dataset.islet.rnaseq$covar[ , -5], reml = TRUE)
```

```{r}
plot_coefCC(x = s1c_asah2, map = map)
```

## Calculate allele effects for many traits near Asah2

```{r}
traits_of_interest <- c("ENSMUSG00000038658", 
                        "ENSMUSG00000024766",
                        "ENSMUSG00000087303", 
                        "ENSMUSG00000087112", 
                        "ENSMUSG00000079344", 
                        "ENSMUSG00000089952", 
                        "ENSMUSG00000052085", 
                        "ENSMUSG00000056078")
expr_for_ae <- expr_chr19[ , colnames(expr_chr19) %in% traits_of_interest]

s1c_out <- list()

for (i in 1:length(traits_of_interest)){
  s1c_out[[i]] <- scan1coef(genoprobs = genoprobs[, 19], pheno = expr_for_ae[ , i], kinship = K$`19`, addcovar = dataset.islet.rnaseq$covar[ , -5], reml = TRUE)
  names(s1c_out)[i] <- traits_of_interest[i]

}
```

```{r}
for (i in 1:length(s1c_out)){
  plot_coefCC(x = s1c_out[[i]], map = map)
}
```

## `get_effects` to extract founder allele effects at each trait's QTL peak position

```{r}
library(qtl2pleio)
```

```{r}
asah2_peak_marker_index <- which(m19 %in% asah2_pos)
asah2_peak_effects <- get_effects(marker_index = 970, allele_effects_matrix = s1c_asah2, map = m19)
```

```{r}
# get peak positions (in Mb) for all traits in traits_of_interest
d1 <- dat[dat$gene_id %in% traits_of_interest, ]
(d2 <- d1[match(x = traits_of_interest, table = d1$gene_id), ])
```

```{r}
# get founder allele effects at each trait's peak position
ae <- list()
for (i in 1:length(traits_of_interest)){
  marker_index <- which(m19 %in% d2$pos[i])
  ae[[i]] <- get_effects(marker_index = marker_index, 
              allele_effects_matrix = s1c_out[[i]], 
              map = m19)
  names(ae)[i] <- names(s1c_out)[i]
}
```

```{r}
asah2_ae <- as.matrix(asah2_peak_effects) %>% 
  t()
rownames(asah2_ae) <- asah2_id
ae6 <- sapply(FUN = function(x)x, X = ae) %>%
  t() %>%
  rbind(asah2_ae)
cbind(cor(t(ae6)) %>% colnames(), c(traits_of_interest, asah2_id))
```


```{r}
traits_of_interest
```

```{r}
(cor(t(ae6)) -> cormat)
```

Which gene expression trait is the least correlated with the others?

```{r}
apply(FUN = prod, X = cormat, MARGIN = 1)
```

It looks like ENSMUSG00000087303 and ENSMUSG00000024766 are the least correlated with the others, while ENSMUSG00000089952 has the highest mean correlation with the other traits. Note also that the minimal (absolute value) entry in the correlation matrix above is between ENSMUSG00000087303 and ENSMUSG00000024766.



