---
title: "Identifying interesting pairs of traits in power study results"
author: "Frederick Boehm"
date: "11/26/2018"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

We examine the first plot, for run 3101, in the file 2018-11-19_power-results.Rmd.

Let's examine the three traits between 30 and 30.25. They have quite distinct LRT values. Two have nearly identical univariate QTL LOD peak heights.

If we believe that the three most important considerations are 1. interlocus distance, 2. univariate assocation strength and 3. similarity of allele effects patterns, then we need to examine the allele effects patterns for these three and Asah2. 

Let's get the annotation information for the three genes of interest (and Asah2).

## Run 3101


```{r}
library(readr)
library(qtl2pleio)
library(tidyverse)
```

```{r}
scans <- list()
fns <- dir("../results/pvl-run3101/", pattern = ".txt")
for (i in 1:length(fns)){
  scans[[i]] <- as_tibble(read.table(file.path("../results/pvl-run3101", fns[i])))
}
lrt <- purrr::map_dbl(.x = scans, .f = calc_lrt_tib)
```

## Parse file names to recover annotation information

We now parse the file names object `fns`, to recover the annotation information for the LRT values.

```{r}
library(stringr)
```

```{r}
gene1 <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[3])
gene2 <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[4]) %>% 
  str_split(pattern = ".txt") %>%
  purrr::map_chr(.f = function(x)x[1])
jobnum <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[2]) %>%
  as.integer()
runnum <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[1]) %>%
  str_remove(pattern = "pvl-run") %>% 
  as.integer()
r3101 <- tibble(filename = fns, gene1 = gene1, gene2 = gene2, run = runnum, job = jobnum, lrt = lrt)
mytib <- r3101
```

```{r}
load("../data/Attie_DO378_eQTL_viewer_v1.Rdata")
```

```{r}
myt <- dataset.islet.rnaseq$annots %>%
  select(gene_id, symbol, middle) %>%
  right_join(mytib, by = c("gene_id" = "gene1")) %>%
  rename(gene1 = gene_id, gene1_symbol = symbol, gene1_middle = middle)
```

We want to add the univariate LOD scores for gene2 values.

```{r}
mytib2 <- dataset.islet.rnaseq$lod.peaks %>% 
  select(- marker.id) %>%
  filter(chrom == 19) %>%
  right_join(myt, by = c("annot.id" = "gene1")) %>%
  rename(gene1 = annot.id, gene1_lod_peak_pos = pos, gene1_lod = lod)
```

## Plot for run 3101

Initially, we ignore allele effects patterns.

```{r}
# get the position for gene2
r3101$gene2[1]
gene2_middle_r3101 <- dataset.islet.rnaseq$annots %>%
  filter(gene_id == r3101$gene2[1]) %>%
  select(middle) %>%
  unlist()
```

```{r}
library(plotly)
```



```{r, out.height = 12, out.width=8}
pp <- mytib2 %>%
  filter(run == 3101) %>%
ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, y = lrt, colour = gene1_lod)) + 
  geom_vline(xintercept = gene2_middle_r3101)
ggplotly(pp)
```


```{r}
mytib2 
```

```{r, out.height = 12, out.width=8}
mytib3 <- dataset.islet.rnaseq$annots %>%
  select(gene_id, middle, symbol) %>%
  right_join(mytib2, by = c("gene_id" = "gene2")) %>%
  rename(gene2 = gene_id, gene2_middle = middle, gene2_symbol = symbol) %>%
  as_tibble()
pp3 <- mytib3 %>%
  ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, y = lrt, colour = gene1_lod)) + facet_grid(rows = vars(gene2_symbol)) + geom_vline(aes(xintercept = gene2_middle))
ggplotly(pp3)
```

```{r}
mytib3 %>%
  filter(gene1_middle < 30.3, gene1_middle > 30)
```

```{r}
ids <- mytib3 %>%
  filter(gene1_middle < 30.3, gene1_middle > 30) %>%
  select(gene1) %>%
  unlist()

```


## Allele effects

```{r}
foo <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 19, lod > 10)

(dat <- dataset.islet.rnaseq$annots %>% 
  filter(chr == 19) %>%
  right_join(foo, by = c("gene_id" = "annot.id")) %>%
  filter(chrom == 19, chr == 19) %>%
  arrange(desc(lod)) %>%
    as_tibble()
 )
asah2_id <- dat %>%
  filter(symbol == "Asah2") %>%
  select(gene_id) %>%
  unlist()
```

```{r}
asah2_pos <- dat %>%
  filter(symbol == "Asah2") %>%
  select(pos) %>%
  unlist()
trait_ids_10mb <- dat %>% 
  filter(pos < asah2_pos + 10, pos > asah2_pos - 10) %>%
  select(gene_id) %>%
  unlist()
```


```{r}
expr_chr19_10mb <- dataset.islet.rnaseq$expr[ , 
                           colnames(dataset.islet.rnaseq$expr) %in% trait_ids_10mb, 
                           drop = FALSE]
```

```{r}
m19 <- map$`19`
m19_tib <- tibble(marker_id = names(m19), position = m19)
which(m19_tib$position > 22)[1]
which(m19_tib$position > 42)[1]
```

## Calculate allele effects for Asah2

```{r}
library(qtl2)
```

```{r}
asah2_expr <- dataset.islet.rnaseq$expr[ , colnames(dataset.islet.rnaseq$expr) == asah2_id, drop = FALSE]
```


```{r}
s1c_asah2 <- scan1coef(genoprobs = genoprobs[, 19], pheno = asah2_expr, kinship = K$`19`, addcovar = dataset.islet.rnaseq$covar[ , -5], reml = TRUE)
```

```{r}
plot_coefCC(x = s1c_asah2, map = map)
```


## Calculate allele effects for all genes of interest

```{r}
s1c_out <- list()
for (i in 1:ncol(expr_chr19_10mb)){
  s1c_out[[i]] <- scan1coef(genoprobs = genoprobs[, 19], 
                            pheno = expr_chr19_10mb[ , i], 
                            kinship = K$`19`, 
                            addcovar = dataset.islet.rnaseq$covar[ , -5], 
                            reml = TRUE
                            )
}
names(s1c_out) <- colnames(expr_chr19_10mb)
```

## Allele effects plots for 3 traits


```{r}
which(names(s1c_out) == ids[1])
plot_coefCC(s1c_out[[which(names(s1c_out) == ids[1])]], map = map)
```

```{r}
plot_coefCC(s1c_out[[which(names(s1c_out) == ids[2])]], map = map)
```

```{r}
plot_coefCC(s1c_out[[which(names(s1c_out) == ids[3])]], map = map)
```

## Calculate the fitted values for each mouse for four traits (at their peak positions)

Our goal now is to calculate the fitted values for all mice and each trait, at the respective trait's QTL peak position.

That is, we'll calculate: 

$$\hat Y = X\hat B$$

First, we get the fitted values for the Asah2 gene at its peak position.

```{r}
which(m19 == asah2_pos) ->index_asah2

asah2_fitted <- cbind(probs[ , , index_asah2], addcovar) %*% s1c_asah2[index_asah2, ]
```

Now, get the fitted values for other 3 traits.

```{r}
helper <- function(x){
  which(m19 == x)
}
  
ids_pos <- mytib3 %>%
  filter(gene1_middle > 30, gene1_middle < 30.3) %>%
  select(gene1_lod_peak_pos) %>%
  unlist()
mt4 <- mytib3 %>%
  filter(gene1_middle > 30, gene1_middle < 30.3)
mt4$gene1_lod_peak_index <- c(helper(ids_pos[1]), helper(ids_pos[2]), helper(ids_pos[3]))
mt4
```

```{r}
uhrf2_fitted <- cbind(probs[ , , mt4$gene1_lod_peak_index[1]], addcovar) %*% s1c_out[[which(names(s1c_out) == ids[1])]][mt4$gene1_lod_peak_index[1], ]

gldc_fitted <- cbind(probs[ , , mt4$gene1_lod_peak_index[2]], addcovar) %*% s1c_out[[which(names(s1c_out) == ids[2])]][mt4$gene1_lod_peak_index[2], ]

mbl2_fitted <- cbind(probs[ , , mt4$gene1_lod_peak_index[3]], addcovar) %*% s1c_out[[which(names(s1c_out) == ids[3])]][mt4$gene1_lod_peak_index[3], ]

```

```{r}
mt4$fitted_values_corr_with_asah2 <- cor(cbind(asah2_fitted, uhrf2_fitted, gldc_fitted, mbl2_fitted))[2:4, 1]
```

```{r}
mt4
```


