---
title: "Initial results from power analyses"
author: "Frederick Boehm"
date: "`r lubridate::now()`"
always_allow_html: yes
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Run 3101


```{r}
library(readr)
library(qtl2pleio)
library(tidyverse)
```

```{r}
scans <- list()
fns <- dir("../results/pvl-run3101/", pattern = ".txt")
for (i in 1:length(fns)){
  scans[[i]] <- as_tibble(read.table(file.path("../results/pvl-run3101", fns[i])))
}
lrt <- purrr::map_dbl(.x = scans, .f = calc_lrt_tib)
```

## Parse file names to recover annotation information

We now parse the file names object `fns`, to recover the annotation information for the LRT values.

```{r}
library(stringr)
```

```{r}
gene1 <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[3])
gene2 <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[4]) %>% 
  str_split(pattern = ".txt") %>%
  purrr::map_chr(.f = function(x)x[1])
jobnum <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[2]) %>%
  as.integer()
runnum <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[1]) %>%
  str_remove(pattern = "pvl-run") %>% 
  as.integer()
r3101 <- tibble(filename = fns, gene1 = gene1, gene2 = gene2, run = runnum, job = jobnum, lrt = lrt)
```






## Run 3102

First, we calculate the LRTs from the scan results.

```{r}
scans <- list()
fns <- dir("../results/pvl-run3102/", pattern = ".txt")
for (i in 1:length(fns)){
  scans[[i]] <- as_tibble(read.table(file.path("../results/pvl-run3102", fns[i])))
}
lrt <- purrr::map_dbl(.x = scans, .f = calc_lrt_tib)
```

Now, we parse the filenames to get the annotation information.

```{r}
gene1 <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[3])
gene2 <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[4]) %>% 
  str_split(pattern = ".txt") %>%
  purrr::map_chr(.f = function(x)x[1])
jobnum <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[2]) %>%
  as.integer()
runnum <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[1]) %>%
  str_remove(pattern = "pvl-run") %>% 
  as.integer()
r3102 <- tibble(filename = fns, gene1 = gene1, gene2 = gene2, run = runnum, job = jobnum, lrt = lrt)
```

Now, combine the two tables before loading the Keller annotation data. We'll load the Keller annotation data so that we can get the gene positions.

```{r}
mytib <- bind_rows(r3101, r3102)
```

```{r}
load("../data/Attie_DO378_eQTL_viewer_v1.Rdata")
```

```{r}
myt <- dataset.islet.rnaseq$annots %>%
  select(gene_id, symbol, middle) %>%
  right_join(mytib, by = c("gene_id" = "gene1")) %>%
  rename(gene1 = gene_id, gene1_symbol = symbol, gene1_middle = middle)
```

We want to add the univariate LOD scores for gene2 values.

```{r}
mytib2 <- dataset.islet.rnaseq$lod.peaks %>% 
  select(- marker.id) %>%
  filter(chrom == 19) %>%
  right_join(myt, by = c("annot.id" = "gene1")) %>%
  rename(gene1 = annot.id, gene1_lod_peak_pos = pos, gene1_lod = lod)
```

## Plot for run 3101

Initially, we ignore allele effects patterns.

```{r}
# get the position for gene2
r3101$gene2[1]
gene2_middle_r3101 <- dataset.islet.rnaseq$annots %>%
  filter(gene_id == r3101$gene2[1]) %>%
  select(middle) %>%
  unlist()
```

```{r}
library(plotly)
```



```{r, out.height = 12, out.width=8}
pp <- mytib2 %>%
  filter(run == 3101) %>%
ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, y = lrt, colour = gene1_lod)) + 
  geom_vline(xintercept = gene2_middle_r3101)
ggplotly(pp)
```


```{r, out.height = 12, out.width=8}
mytib3 <- dataset.islet.rnaseq$annots %>%
  select(gene_id, middle, symbol) %>%
  right_join(mytib2, by = c("gene_id" = "gene2")) %>%
  rename(gene2 = gene_id, gene2_middle = middle, gene2_symbol = symbol) %>%
  as_tibble()
pp3 <- mytib3 %>%
  ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, y = lrt, colour = gene1_lod)) + facet_grid(rows = vars(gene2_symbol)) + geom_vline(aes(xintercept = gene2_middle))
ggplotly(pp3)
```

## Truncated LRT plots

```{r}
pp4 <- mytib3 %>%
  mutate(lrt_trunc = pmin(20, .$lrt)) %>%
  ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, y = lrt_trunc, colour = gene1_lod)) + facet_grid(rows = vars(gene2_symbol)) + geom_vline(aes(xintercept = gene2_middle))
ggplotly(pp4)
```

## Allele effects

We now calculate allele effects on chromosome 19 for all traits in our study.

We first need to use the annotation object to get a list of trait ids that are in our study.

```{r}
foo <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 19, lod > 10)

(dat <- dataset.islet.rnaseq$annots %>% 
  filter(chr == 19) %>%
  right_join(foo, by = c("gene_id" = "annot.id")) %>%
  filter(chrom == 19, chr == 19) %>%
  arrange(desc(lod)) %>%
    as_tibble()
 )
asah2_id <- dat %>%
  filter(symbol == "Asah2") %>%
  select(gene_id) %>%
  unlist()
```

```{r}
asah2_pos <- dat %>%
  filter(symbol == "Asah2") %>%
  select(pos) %>%
  unlist()
trait_ids_10mb <- dat %>% 
  filter(pos < asah2_pos + 10, pos > asah2_pos - 10) %>%
  select(gene_id) %>%
  unlist()
```


```{r}
expr_chr19_10mb <- dataset.islet.rnaseq$expr[ , 
                           colnames(dataset.islet.rnaseq$expr) %in% trait_ids_10mb, 
                           drop = FALSE]
```

```{r}
m19 <- map$`19`
m19_tib <- tibble(marker_id = names(m19), position = m19)
which(m19_tib$position > 22)[1]
which(m19_tib$position > 42)[1]
```

## Calculate allele effects for Asah2

```{r}
library(qtl2)
```

```{r}
asah2_expr <- dataset.islet.rnaseq$expr[ , colnames(dataset.islet.rnaseq$expr) == asah2_id, drop = FALSE]
```


```{r}
s1c_asah2 <- scan1coef(genoprobs = genoprobs[, 19], pheno = asah2_expr, kinship = K$`19`, addcovar = dataset.islet.rnaseq$covar[ , -5], reml = TRUE)
```

```{r}
plot_coefCC(x = s1c_asah2, map = map)
```


## Calculate allele effects for all genes of interest

```{r}
s1c_out <- list()
for (i in 1:ncol(expr_chr19_10mb)){
  s1c_out[[i]] <- scan1coef(genoprobs = genoprobs[, 19], 
                            pheno = expr_chr19_10mb[ , i], 
                            kinship = K$`19`, 
                            addcovar = dataset.islet.rnaseq$covar[ , -5], 
                            reml = TRUE
                            )
}
names(s1c_out) <- colnames(expr_chr19_10mb)
```

## `get_effects` to extract founder allele effects at each trait's QTL peak position

```{r}
asah2_peak_marker_index <- which(m19 %in% asah2_pos)
asah2_peak_effects <- get_effects(marker_index = asah2_peak_marker_index, allele_effects_matrix = s1c_asah2, map = m19)
```


Define `traits_of_interest` as those in the `expr_chr19_10mb` object.

```{r}
colnames(expr_chr19_10mb) -> traits_of_interest
```




```{r}
# get peak positions (in Mb) for all traits in traits_of_interest
d1 <- dat[dat$gene_id %in% traits_of_interest, ]
(d2 <- d1[match(x = traits_of_interest, table = d1$gene_id), ])
```

```{r}
# get founder allele effects at each trait's peak position
ae <- list()
for (i in 1:length(traits_of_interest)){
  marker_index <- which(names(m19) %in% d2$nearest.marker.id[i])
  ae[[i]] <- get_effects(marker_index = marker_index, 
              allele_effects_matrix = s1c_out[[i]], 
              map = m19)
  names(ae)[i] <- names(s1c_out)[i]
}
```

```{r}
asah2_ae <- as.matrix(asah2_peak_effects) %>% 
  t()
rownames(asah2_ae) <- asah2_id
ae6 <- bind_rows(ae)
```


```{r}
traits_of_interest
```

```{r}
(cor(ae6) -> cormat)
```

## Add allele effects correlations to `mytib3`

```{r}
mt4 <- mytib3 %>%
  mutate(corr_with_anchor = apply(FUN = function(x)cormat[rownames(cormat) == x[1], colnames(cormat) == x[4]],
      MARGIN = 1,
      X = mytib3
  )
  )
```

## Truncated LRT plots

```{r}
pp5 <- mt4 %>%
  mutate(lrt_trunc = pmin(20, .$lrt)) %>%
  ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, 
                           y = lrt_trunc, 
                           colour = abs(corr_with_anchor),
                           alpha = gene1_lod)) + facet_grid(rows = vars(gene2_symbol)) + geom_vline(aes(xintercept = gene2_middle))
ggplotly(pp5)
```


## LRT plots, no truncation

```{r}
pp5 <- mt4 %>%
  mutate(lrt_trunc = pmin(20, .$lrt)) %>%
  ggplot() + 
  geom_point(mapping = aes(x = gene1_middle, 
                           y = lrt, 
                           colour = abs(corr_with_anchor),
                           alpha = gene1_lod)) + facet_grid(rows = vars(gene2_symbol)) + geom_vline(aes(xintercept = gene2_middle))
ggplotly(pp5)
```

## Correlations of 3 factors with LRT


```{r}
cor(abs(mt4$corr_with_anchor), mt4$lrt)
```

```{r}
ggplot(mt4) + geom_point(aes(y = lrt, x = abs(corr_with_anchor)))
```




```{r}
cor(mt4$gene1_lod, mt4$lrt)
```

```{r}
ggplot(mt4) + geom_point(aes(y = lrt, x = gene1_lod))
```




```{r}
mt5 <- mt4 %>%
  mutate(inter_locus_dist = abs(gene2_middle - gene1_middle))
cor(mt5$inter_locus_dist, mt5$lrt)
```

```{r}
ggplot(mt5) + geom_point(aes(y = lrt, x = inter_locus_dist))
```






