---
title: "Creating figures with LRTs"
author: "Frederick Boehm"
date: "12/4/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

First, we read in the four tibbles from csv files.

```{r}
library(tidyverse)
```

```{r}
read_csv("2018-11-26_lrt-tibble.csv") -> r3104
read_csv("2018-12-03_lrt-tibble-run3105.csv") -> r3105
read_csv("2018-12-03_lrt-tibble-run3106.csv") -> r3106
read_csv("2018-12-03_lrt-tibble-run3107.csv") -> r3107
(bind_rows(r3104, r3105, r3106, r3107) -> rr)
```

Next, load the Attie data set from Data Dryad.

```{r}
load("../data/Attie_DO378_eQTL_viewer_v1.Rdata")
```


We then work with the annotations object.

```{r}
foo <- dataset.islet.rnaseq$annots %>%
  select(gene_id, symbol, start, end, middle) 
rr2 <- foo %>% 
  right_join(rr, by = c("gene_id" = "gene1")) %>%
  rename(gene1_symbol = symbol, gene1 = gene_id, gene1_start = start, gene1_end = end, gene1_middle = middle) %>%
  left_join(foo, by = c("gene2" = "gene_id")) %>%
  rename(gene2_symbol = symbol, gene2_start = start, gene2_end = end, gene2_middle = middle)
```

We need to add additional information from univariate QTL analyses for both gene1 and gene2. 




```{r}
bar <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == 19)
rr3 <- bar %>%
  right_join(rr2, by = c("annot.id" = "gene1")) %>%
  rename(gene1 = annot.id, gene1_marker.id = marker.id, gene1_pos = pos, gene1_lod = lod) %>%
  left_join(bar, by = c("gene2" = "annot.id")) %>%
  rename(gene2_marker.id = marker.id, gene2_pos = pos, gene2_lod = lod) %>%
  select(- chrom.x, - chrom.y)
```

## LRT v position

```{r}
library(plotly)
```

```{r}
(pp <- rr3 %>%
  ggplot() + geom_point(aes(x = gene1_middle, y = lrt, colour = gene1_lod)) + geom_vline(aes(xintercept = gene2_middle)) + xlab("Chromosome 19 position") + ylab("Pleiotropy v separate QTL test statistic") + facet_grid(rows = gene2_symbol ~ .)
)

```

```{r}
ggplotly(pp)
```

## Arrange the panels by gene2 lod score

```{r}
(nodup <- rr3 %>%
  filter(!duplicated(gene2)) %>%
  arrange(desc(gene2_lod)))
```

```{r}
rr3 %>%
  mutate(gene2_symbol_factor = factor(gene2_symbol, levels = nodup$gene2_symbol, labels = paste0(nodup$gene2_symbol, " (", round(nodup$gene2_lod, 1), ")"))) %>%
  ggplot() + geom_point(aes(x = gene1_middle, y = lrt, colour = gene1_lod)) + geom_vline(aes(xintercept = gene2_middle)) + xlab("Chromosome 19 position") + ylab("Pleiotropy v separate QTL test statistic") + facet_grid(rows = gene2_symbol_factor ~ .)
ggsave(filename = "lrt-v-middle-of-gene.jpg", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-middle-of-gene.svg", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-middle-of-gene.eps", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-middle-of-gene.pdf", width = 7, height = 7, units = "in" )

```

## LRT v univariate LOD by anchor gene


```{r}
rr3 %>%
  mutate(gene2_symbol_factor = factor(gene2_symbol, levels = nodup$gene2_symbol, labels = paste0(nodup$gene2_symbol, " (", round(nodup$gene2_lod, 1), ")"))) %>%
  ggplot() + geom_point(aes(x = gene1_lod, y = lrt, colour = gene1_lod)) + xlab("Univariate LOD") + ylab("Pleiotropy v separate QTL test statistic") + facet_grid(rows = gene2_symbol_factor ~ .)
ggsave(filename = "lrt-v-univariate-lod.jpg", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-univariate-lod.svg", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-univariate-lod.eps", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-univariate-lod.pdf", width = 7, height = 7, units = "in" )

```


## LRT v correlation (between fitted values)

We want to include our covariates in the fitted values calculations.

We'll use the function `qtl2::fit1`.

```{r}
library(qtl2)
```

`rr3$gene1_marker.id` tells us the name of the marker for gene1's LOD peak. So, we'll use it to get the matrix of allele dosages for input to `fit1`. Note that the third list element in the output of `dimnames` gives the marker_id values

```{r}
dimnames(genoprobs$`19`)[[3]]
```

Let's read in the expr data for the 80 traits of interest.

```{r}
readRDS("../data/chr19_expr_10mb.rds") -> expr
colnames(expr)
```

```{r}
ann80 <- rr3 %>%
  select(gene1, gene1_marker.id, gene1_symbol) %>%
  filter(!duplicated(gene1_symbol))
fit1_out <- list()
for (i in 1:80){
  pr <- genoprobs$`19`[ , , which(dimnames(genoprobs$`19`)[[3]] == ann80$gene1_marker.id[i])]
  fit1_out[[i]] <- fit1(genoprobs = pr, 
       pheno = expr[, which(colnames(expr) == ann80$gene1[i]), drop = FALSE], 
       kinship = K$`19`, 
       addcovar = dataset.islet.rnaseq$covar, 
       reml = TRUE
       )
}
```

```{r}
sapply(FUN = function(x)x$fitted, X = fit1_out) -> fits
colnames(fits)
colnames(fits) <- ann80$gene1
rownames(fits)
```


```{r}
cor(fits) -> fitcor
```

## Create and populate the `fit1_corr` with correlations

```{r}
rr3$fit1_corr <- NA
for (i in 1:nrow(rr3)){
  rr3$fit1_corr[i] <- fitcor[which(rownames(fitcor) == rr3$gene1[i]), which(colnames(fitcor) == rr3$gene2[i])]
}
```




```{r}
rr3 %>%
  mutate(gene2_symbol_factor = factor(gene2_symbol, levels = nodup$gene2_symbol, labels = paste0(nodup$gene2_symbol, " (", round(nodup$gene2_lod, 1), ")"))) %>%
  ggplot() + geom_point(aes(x = fit1_corr, y = lrt, colour = gene1_lod)) + xlab("Fitted values correlation") + ylab("Pleiotropy v separate QTL test statistic") + facet_grid(rows = gene2_symbol_factor ~ .)
ggsave(filename = "lrt-v-corr.jpg", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-corr.svg", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-corr.eps", width = 7, height = 7, units = "in" )
ggsave(filename = "lrt-v-corr.pdf", width = 7, height = 7, units = "in" )

```

## Allele effects plots

We now create founder allele effects plots for the four anchor genes: Asah2, Lipo1, Lipo2, and the fourth gene.

We first need the Ensembl ids for the four genes.

```{r}
unique(rr2$gene2) -> id4
unique(rr2$gene2_symbol) -> id4_symbols
expr4 <- expr[ , colnames(expr) %in% id4]
```

```{r}
out4 <- list()
for (i in 1:4){
  out4[[i]] <- scan1coef(genoprobs = list(genoprobs$`19`), 
            pheno = expr4[ , i, drop = FALSE], 
            kinship = K$`19`, 
            addcovar = dataset.islet.rnaseq$covar, 
            reml = TRUE)
}
```



```{r}
for (i in 1:4){
  if (i == 1){
    leg <- "bottomright"
  } else {
    leg <- NULL
  }
  uni_peak_pos <- nodup %>%
    filter(gene2_symbol == id4_symbols[i]) %>%
    select(gene2_pos) %>%
    unlist()
  epsfn <- paste0("allele_effects_", id4_symbols[i], ".eps")
  pdffn <- paste0("allele_effects_", id4_symbols[i], ".pdf")
  setEPS()
  postscript(epsfn, 
             width = 6, 
             height = 4
             )
  plot_coefCC(out4[[i]], 
              map = map[19], 
              legend = leg, 
              ylim = c(- 2.5, 2.5), 
              legend_ncol = 4
              )
  text(x = 50, 
       y = 2, 
       labels = id4_symbols[i]
       )
  #abline(v = uni_peak_pos)
  dev.off()
  pdf(pdffn, 
      width = 6, 
      height = 4
      )
  cex_all <- 0.75
  par(cex = cex_all)
  plot_coefCC(out4[[i]], 
              map = map[19], 
              legend = leg, 
              ylim = c(- 2.5, 3), 
              legend_ncol = 4, 
              )
  text(x = 50, 
       y = 2, 
       labels = id4_symbols[i],
       cex = 1.5 / cex_all
       )
  #abline(v = uni_peak_pos)
  dev.off()
}
```

## Make tables for inclusion in .tex file

```{r}
library(xtable)
```

We make a xtable for the 

```{r, results = "asis"}
nodup %>%
  select(gene2_symbol, gene2_start, gene2_end, gene2_pos, gene2_lod) %>%
  rename(symbol = gene2_symbol, start = gene2_start, end = gene2_end, peak_position = gene2_pos, lod = gene2_lod) %>% 
  xtable() %>%
  print.xtable(include.rownames = FALSE)
```

Make the table for the 76 non-anchor genes.

```{r}
rr3 %>%
  select(gene1_symbol, gene1_start, gene1_end, gene1_pos, gene1_lod) %>%
  rename(symbol = gene1_symbol, start = gene1_start, end = gene1_end, peak_position = gene1_pos, lod = gene1_lod) %>%
  filter(!duplicated(symbol)) %>%
  filter(!(symbol %in% c("Asah2", "Lipo1", "Lipo2", "4933413C19Rik"))) %>%
  arrange(desc(lod))%>% 
  xtable() %>%
  print.xtable(include.rownames = FALSE, size = "\\tiny")
```



