---
title: "Assembling scan results"
author: "Frederick Boehm"
date: "11/25/2018"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Overview

Our goal here is to assemble the scan results for run 3103, which is part of the chromosome 19 power study. Recall that we split each 2d scan (of 1000 by 1000 markers) into 25 condor jobs. We now need to assemble the 25 results files per scan into a single R object.

## Diagnostics

First, let's ensure that we have the correct number of files. 25 jobs failed. This was due to the fact that each of 80 expression traits was paired with Asah2 gene expression, and Asah2 expression itself was among the 80... so all 25 of its jobs failed.

So, we should have 79 * 25 jobs.

```{bash}
ls ../results/pvl-run3104/*.txt | wc -l
```

We see that there are 1975 files, as needed.



```{r}
library(tidyverse)
```


## Read results files

```{r}
res <- "../results/pvl-run3104"
dir(res) -> fns
results <- list()
for (i in 1:1975){
  results[[i]] <- read.table(file.path(res, fns[i]), stringsAsFactors = FALSE) %>%
    as_tibble() %>%
    mutate(filename = fns[i])
}
```

```{r}
results2 <- results %>%
  bind_rows()
```


```{r}
foo <- str_split(fns, pattern = "_") %>%
  purrr::map_chr(.f = function(x)x[3])
results2$gene1 <- rep(foo, each = 40000)
```


## "Create 79 tibbles" by gene1 values

```{r}
library(qtl2pleio)
```


```{r}
gene1_79 <- unique(results2$gene1)
lrt <- numeric(length = 79)
for (i in 1:79){
  lrt[i] <- results2 %>%
    filter(gene1 == gene1_79[i]) %>%
    select(1:3) %>%
    calc_lrt_tib()
}
tibble(gene1 = gene1_79, lrt = lrt)
```

```{r}
write_csv(x = tibble(gene1 = gene1_79, lrt = lrt), path = "2018-11-26_lrt-tibble.csv")
```



